human_analysis/nbconverted/1_process_recount2_data.py

```diff
@@ -0,0 +1,389 @@
+# Map ensembl ids with version number to gene_id_mapping_df
+gene_id_mapping = pd.merge(gene_id_mapping, 
+                           ensembl_gene_ids, 
+                           left_on='ensembl_gene_id',
+                           right_on='ensembl_parsed', 
+                           how='outer')
+
+print(gene_id_mapping.shape)
+gene_id_mapping.set_index('ensembl_version', inplace=True)
+gene_id_mapping.head()
+
+
+# Since this experiment contains both RNA-seq and smRNA-seq samples which are in different ranges so we will drop smRNA samples so that samples are within the same range. The analysis identifying these two subsets of samples can be found in this [notebook](0_explore_input_data.ipynb)
+
+# In[17]:
+
+
+# Replace ensembl ids with gene symbols
+template_data = process.replace_ensembl_ids(template_data,
+                                            gene_id_mapping)
+
+
+# In[18]:
+
+
+template_data.head()
+
+
+# In[19]:
+
+
+# Get intersection of gene lists
+our_gene_ids_hgnc = template_data.columns
+shared_genes_hgnc = list(set(our_gene_ids_hgnc).intersection(published_generic_genes))
+print(len(shared_genes_hgnc))
+
+
+# In[20]:
+
+
+# Save shared genes
+shared_genes_file = os.path.join(
+    local_dir,
+    "shared_gene_ids.pickle")
+
+outfile = open(shared_genes_file,'wb')
+pickle.dump(shared_genes_hgnc,outfile)
+outfile.close()
+
+
+# In[21]:
+
+
+# Drop smRNA samples so that samples are within the same range
+smRNA_samples = ["SRR493961",
+                 "SRR493962",
+                 "SRR493963",
+                 "SRR493964",
+                 "SRR493965",
+                 "SRR493966",
+                 "SRR493967",
+                 "SRR493968",
+                 "SRR493969",
+                 "SRR493970",
+                 "SRR493971",
+                 "SRR493972"]
+
+
+# In[22]:
+
+
+# Drop samples
+template_data = template_data.drop(smRNA_samples)
+
+
+# In[23]:
+
+
+# Drop genes
+template_data = template_data[shared_genes_hgnc]
+
+print(template_data.shape)
+template_data.head()
+
+
+# In[24]:
+
+
+print(len(template_data.columns) - len(shared_genes_hgnc))
+
+
+# *Note:* There is a difference in the number of `shared_genes_hgnc` and genes in the template experiment because 3 genes have 2 different ensembl gene ids have map to the same hgnc symbol (one forward, one reverse)
+
+# In[25]:
+
+
+# Save 
+template_data.to_csv(template_data_file, float_format='%.5f', sep='\t')
+
+
+# ### Normalize compendium 
+
+# In[26]:
+
+
+# Load real gene expression data
+original_compendium_file = params['compendium_data_file']
+
+
+# In[27]:
+
+
+# Read data
+original_compendium = pd.read_table(
+    original_compendium_file,
+    header=0,
+    sep='\t',
+    index_col=0)
+
+print(original_compendium.shape)
+original_compendium.head()
+
+
+# In[28]:
+
+
+# Replace ensembl ids with gene symbols
+#original_compendium = process.replace_ensembl_ids(original_compendium,
+#                                                gene_id_mapping)
```

Comment:
Can be removed

Reply:
This is another comment that is temporarily commented for the purposes of debugging but will need to be uncommented later